core/src/com/mygdx/game/config/GameConfigs.java

```diff
@@ -1,74 +1,48 @@
 package com.mygdx.game.config;
 
 import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.LongMap;
 import lombok.NonNull;
 
-import java.util.HashMap;
-import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
 
+@Singleton
 public class GameConfigs {
 
-  @NonNull
-  private final Map<Class<? extends EntityConfig>, Map<Integer, EntityConfig>> entityConfigMap;
+  private final LongMap<EntityConfig> entityConfigMap = new LongMap<>();
 
+  @Inject
   public GameConfigs() {
-    entityConfigMap = new HashMap<>();
   }
 
   public <T extends EntityConfig> T get(@NonNull final Class<T> entityClass,
```

Comment:
Jaki masz pomysł na znajomość tego, które id to które entity? Gdzieś to będziemy zapisywać? Czy da się to jakoś wyciągać?

Reply:
Ogólnie to jak mamy tutaj EntityConfig to on ma w sobie wspólne pole dla każdego EntityConfigu - Id. To pole jest zapełniane z jsona, które są wspólne dla serwera i dla klienta. 

Kod klienta ma być na tyle elastyczny, żeby to Id starczało do pobrania odpowiedniego entityconfiga i wywołania fabryki, która sobie już poradzi, żeby dodać jakieś grafiki czy nazwy. Jeśli chcemy rozróżniać to jeszcze, tak ogólnikowo bez kontekstu, to można się umówić, że wartości od tylu do tylu są zarezerwowane na cośtam, a od tylu do tylu na cośtam, ale wydaje mi się, że to nie będzie aż tak potrzebne i tylko przy tworzeniu entity w fabrykach trzeba dodać odpowiednie componenty (np. moveable).

Reply:
Okej przekonuje mnie to